backend/fms_core/services/experiment_run.py

```diff
@@ -0,0 +1,170 @@
+from datetime import datetime
+from ..models import (
+    ExperimentRun,
+    Container,
+    Process,
+    ProcessMeasurement,
+    PropertyValue,
+    Sample,
+    SampleLineage
+)
+from ..utils import (
+    blank_str_to_none,
+    float_to_decimal,
+    get_normalized_str,
+)
+
+def create_experiment_run(experiment_type, instrument, container, start_date,
+                 sample_rows_for_experiment, properties,
+                 protocols_objs_dict, property_types_objs_dict):
+    errors = {}
+    experiment_run = None
+
+    parent_process, processes_by_protocol_id = create_processes_from_protocols(protocols_objs_dict)
+
+    try:
+        experiment_run = ExperimentRun.objects.create(experiment_type=experiment_type,
+                                     instrument=instrument,
+                                     container=container,
+                                     process=parent_process,
+                                     start_date=start_date)
+
+        print('SERVICES - experiment_run: ', experiment_run)
+
+        result = create_properties(properties, property_types_objs_dict, processes_by_protocol_id)
+        if result['errors'] != {}:
+            errors['properties'] = result['errors']
+
+        print('SERVICES - properties/result: ', result)
+
+        result = handle_samples(experiment_run, sample_rows_for_experiment)
+        if result['errors'] != {}:
+            errors['samples'] = result['errors']
+
+        print('SERVICES - samples/result: ', result)
+
+    except Exception as e:
+        errors['experiment_run'] = e
+        print('SERVICES - experiment_run/exception: ', e)
+
+
+    return {'object': experiment_run, 'errors': errors}
+
+def create_processes_from_protocols(protocols_objs_dict):
+    processes_by_protocol_id = {}
+    for protocol in protocols_objs_dict.keys():
+        parent_process = Process.objects.create(protocol=protocol, comment="")
+        for subprotocol in protocols_objs_dict[protocol]:
+            sp = Process.objects.create(protocol=subprotocol,
+                                        parent_process=parent_process,
+                                        comment="Experiment")
+            processes_by_protocol_id[subprotocol.id] = sp
+
+    return (parent_process, processes_by_protocol_id)
```

Comment:
What happens if there is 2 parent processes ?

Reply:
could be renamed to top_process... an experiment run has only 1 process, so it must be the top_process